### DISCUSSION TITLE
Complex indoor scene have limited ray numbers

### DISCUSSION THREAD

**Discussion by ZitongLan** [NVlabs/sionna#359] [General]
[room.zip](https://github.com/NVlabs/sionna/files/14609603/room.zip)
Hi, I am woking on a customize indoor scene with real indoor scene scan through Lidar. I convert the lidar data into mesh data and thus get .ply file, make the whole scene as a uniform material in blender. But when I do ray tracing, I only get limited number of reflection path. (Even though I have set the max depth up to 16 and ray number to 3e6). like this 
![image](https://github.com/NVlabs/sionna/assets/57742620/e7161902-791b-4f75-804f-0232a635114d)

Should I further simplify the geometry like mesh, before I load them into the blender? Or should I merge more faces together so that they would have similar surface normal? I am kind of confuse here. Can you give me some advice on solving this problem?

**Comment by jhoydis** [comment]
Hi @ZitongLan,

The issue with your scene is that it is too detailed so that there are simply too many primitives that need to be tested for specular reflections. Sionna RT only tests primitives that are actually hit by a ray. 

You could (1) try to shoot far more rays (if your GPU has the memory), (2) simplify the scene by combining primitives with similar normals, and (3) add diffuse reflections (scattering). Keep in mind that you would need to configure the scattering coefficient for all objects to see these effects.

Hope that helps.
